process_report/process_report.py

```diff
@@ -347,21 +367,50 @@ def apply_credits_new_pi(dataframe, old_pi_file):
     dataframe[CREDIT_CODE_FIELD] = None
     dataframe[BALANCE_FIELD] = Decimal(0)
 
-    old_pi_dict = load_old_pis(old_pi_file)
+    old_pi_df = load_old_pis(old_pi_file)
 
-    current_pi_list = dataframe[PI_FIELD].unique()
+    current_pi_set = set(dataframe[PI_FIELD])
     invoice_month = dataframe[INVOICE_DATE_FIELD].iat[0]
+    invoice_pis = old_pi_df[old_pi_df[PI_FIRST_MONTH] == invoice_month]
+    if invoice_pis["Initial Credits"].empty or pandas.isna(
+        new_pi_credit_amount := invoice_pis["Initial Credits"].iat[0]
+    ):
+        new_pi_credit_amount = 1000
```

Comment:
You've already defined `new_pi_credit_amount` as 1000 at the top of the function. Either use a different variable name to prevent overwriting it, or do not set this variable to a different value if there is not a number in that column.